core/xmake.lua

```diff
@@ -65,15 +65,23 @@ option_end()
 option("readline")
     set_description("Enable or disable readline library")
     add_links("readline")
-    add_cincludes("readline/readline.h")
+    add_cincludes("stdio.h", "readline/readline.h")
     add_cfuncs("readline")
     add_defines("XM_CONFIG_API_HAVE_READLINE")
 option_end()
 
 -- the curses option
 option("curses")
     set_description("Enable or disable curses library")
-    add_links("curses")
+    if is_plat("mingw") then
```

Comment:
we cannot use is_plat and get_config in option description scope. it does not work.

Reply:
we should handle it in before_check

https://github.com/tboox/tbox/blob/c6b0a56076941b8263e162c7fe7b0870ea44e09c/src/xmake.lua#L90